game/plugin/skills/woodcutting/src/woodcutting.plugin.kts

```diff
@@ -0,0 +1,117 @@
+import org.apollo.cache.def.ItemDefinition
+import org.apollo.game.action.AsyncDistancedAction
+import org.apollo.game.message.impl.ObjectActionMessage
+import org.apollo.game.model.Position
+import org.apollo.game.model.World
+import org.apollo.game.model.entity.Player
+import org.apollo.game.model.entity.Skill
+import org.apollo.game.model.entity.obj.GameObject
+import org.apollo.game.plugin.skills.woodcutting.*
+import org.apollo.game.plugins.api.Definitions
+import org.apollo.game.plugins.api.woodcutting
+import org.apollo.game.plugins.api.skills
+import org.apollo.game.scheduling.ScheduledTask
+import java.util.*
+
+class WoodcuttingTarget(val objectId: Int, val position: Position, val wood: Wood) {
+
+    fun getObject(world: World): Optional<GameObject> {
+        val region = world.regionRepository.fromPosition(position)
+        val obj = region.findObject(position, objectId)
+
+        return obj
+    }
+
+    fun isSuccessful(mob: Player): Boolean {
```

Comment:
I think this must be influenced by the `Player`'s wc level in some way - if this is the case, can we add a TODO?

Reply:
This is the correct behavior. The player gets logs faster from a tree as his level goes up, but the tree always has the same chance at being cut down. Some trees will always get cut down and only yield one logs such as NORMAL, but other such as OAK will usually yield multiple logs before being cut down. http://oldschoolrunescape.wikia.com/wiki/Woodcutting#Trivia

Reply:
Sorry, I thought this function was for gathering logs. Could we rename it to `fellsTree` or something?

Reply:
Yeah, I already renamed it,